### ISSUE TITLE
Faulty url normalization causes an infinite redirect loop

### ISSUE THREAD

**Issue by jdolearydl** [AzureAD/microsoft-authentication-library-for-js#2705]
## Library
- [x ] `@azure/msal-browser@2.7.0`

## Framework
N/A

## Description
ClientApplication.ts can cause an infinite redirect loop when loginRequestUrlNormalized doesn't equal currentUrlNormalized due to query string `?`.
For example: `"http://localhost:8000/observation/field?" !== "http://localhost:8000/observation/field"`

When ClientApplication.ts line 179
`if (loginRequestUrlNormalized === currentUrlNormalized && this.config.auth.navigateToLoginRequestUrl) {`
is false due to this mistaken mismatch, it eventually goes into line 208
                `await BrowserUtils.navigateWindow(loginRequestUrl, this.config.system.redirectNavigationTimeout, this.logger, true);` which navigates back to loginRequestUrl which starts the process all over and creates an infinite redirect loop.

This all happens because calling `UrlString.removeHashFromUrl('http://localhost:8000/observation/field?')` returns `'http://localhost:8000/observation/field?/'` instead of `'http://localhost:8000/observation/field/'`.
If removeHashFromUrl or possibly UrlString.canonicalizeUri were to also remove the queryString this bug would be avoided.

## Error Message
None

## MSAL Configuration
```js
{
    auth: {
        clientId: redacted,
        authority: redacted,
        redirectUri: "http://localhost:8000/login"
    },
    cache: {
        cacheLocation: "localStorage",
        storeAuthStateInCookie: false,
    }
}
```

## Reproduction steps

Simple way: write a test to prove that removeHashFromUrl will turn `'http://localhost:8000/foo?'` into `'http://localhost:8000/foo?/` which is obviously not the same url.

Full Reproduction: Start the **redirect flow** from a page such as `http://localhost:8000/observation/field?` which also calls  `OAuth.MSAL.handleRedirectPromise()`.  The `?` of the query string will cause the normalized urls to mismatch and create an infinite loop. (Note: it doesn't make a difference that the config redirectUri is not the same as the ORIGIN_URI)

## Expected behavior
It should sign the user in, not enter into an infinite redirect loop

## Browsers/Environment
- [x] Chrome
- [x] Firefox
- [x] Edge
- [ ] Safari
- [ ] IE
- [ ] Other (Please add browser name here)

**Comment by jdolearydl** [comment]
![msal-loop-demo](https://user-images.githubusercontent.com/43141484/101391078-d501a400-3891-11eb-8e72-e55afe4cb1a7.gif)

**Comment by tnorling** [comment]
@jdolearydl Thanks for bringing this to our attention. It looks like when the query string is empty our `clearHash` function is stripping the ? off the current url which is what is preventing the 2 from ever being equal in this scenario and causing the loop. I will put up a PR to address this shortly but in the meantime there are a couple different workarounds you can try:
1. Set the `redirectStartPage` parameter in your request passed to `loginRedirect` to the url without the empty query string
2. Set your query string to anything other than empty before calling `loginRedirect`
3. Remove the ? before calling `loginRedirect`

**Comment by jdolearydl** [comment]
I'll give that a shot, thanks!

**Comment by tnorling** [comment]
@jdolearydl Fix will be available in version 2.8.0 which will be released shortly
